Feature added: Ability to force email and name entry

bkruger99 opened the pull request:
Only available via lambda at the moment as I'm not sure how to parse .env file for the front end ui to do the right thing.

bkruger99 committed:
Add ability to force email and real name fields via env file.

TODO:
apply these changes to frontend ui as well.

bkruger99 committed:
Merge pull request #1 from bkruger99/force_fields

Add ability to force email and real name fields via env file.

jimpick commented:
This looks great!

When I get some free time (this weekend hopefully), I'll try it out and merge it in if all goes well.

bkruger99 commented:
Thanks.  If you have an idea how to inject into the UI to read the env variables (js is not my strong suit), then I can implement before submission on the UI side.    Tests maybe broken, didn't update those, will try to get a look at it, but don't wait for me. :)

jimpick commented:
I think we'll need to update this file to include the env variable and
compile it into the bundle that webpack generates:

https://github.com/jimpick/lambda-comments/blob/master/packages/frontend/webpack.config.js#L69

I haven't looked at your code yet - I'll check it out soon.

On Wed, Jul 27, 2016 at 5:31 PM, Brian Kruger notifications@github.com
wrote:

> Thanks. If you have an idea how to inject into the UI to read the env
> variables (js is not my strong suit), then I can implement before
> submission on the UI side. Tests maybe broken, didn't update those, will
> try to get a look at it, but don't wait for me. :)
> 
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> https://github.com/jimpick/lambda-comments/pull/14#issuecomment-235763366,
> or mute the thread
> https://github.com/notifications/unsubscribe-auth/AABRlsfB-tCFHMS0Dtq4EkirqSjEsihrks5qZ_hdgaJpZM4JWqUz
> .

bkruger99 commented:
If the .env can be pulled into the FE UI, then adding your other open issue of content length should be fairly trivial.  Whenever you get around to the code check, no hurry.

bkruger99 commented:
Looking at this a little more, yeah. I get how to potentially pass some things into the UI.  Probably would be useful to make a util/config.js perhaps to pass some items in.  Not sure if there's a way to stringify everything from it in case of major expansion, but I have an idea.  I'll play around and try to amend the pull request to cover front end ui, probably be a few days before I get to it.

bkruger99 committed:
add frontend validation before lambda post
added content length and (mostly) blank post checking
added bonus - comment box counter

bkruger99 committed:
Merge pull request #2 from bkruger99/force_fields

add frontend validation before lambda post

bkruger99 commented:
Finally got around to finishing this as it was bugging me.   

Front end validation done (not in the tests)
Implemented #1 and #8 fixes as well both front and back end. 
(#8 is a simple regex)

Added bonus: character counter above the text box.

bkruger99 commented:
@jimpick you still alive there? :)

jimpick commented:
Sorry, been super busy. I'm way overdue to put some time in on this project.
